spyder/plugins/editor/widgets/completion.py

```diff
@@ -0,0 +1,407 @@
+
+        # If update_current called close, stop loading
+        if not self.completion_list:
+            return
+
+        # If only one, must be chosen if not automatic
+        single_match = self.count() == 1
+        if single_match and not self.automatic:
+            self.item_selected()
+            self.hide()
+            # signal used for testing
+            self.sig_show_completions.emit(completion_list)
+            return
+
+        self.show()
+        self.setFocus()
+        self.raise_()
+
+        # Retrieving current screen height
+        desktop = QApplication.desktop()
+        srect = desktop.availableGeometry(desktop.screenNumber(self))
+        screen_right = srect.right()
+        screen_bottom = srect.bottom()
+
+        point = self.textedit.cursorRect().bottomRight()
+        point = self.textedit.calculate_real_position(point)
+        point = self.textedit.mapToGlobal(point)
+
+        # Computing completion widget and its parent right positions
+        comp_right = point.x() + self.width()
+        ancestor = self.parent()
+        if ancestor is None:
+            anc_right = screen_right
+        else:
+            anc_right = min([ancestor.x() + ancestor.width(), screen_right])
+
+        # Moving completion widget to the left
+        # if there is not enough space to the right
+        if comp_right > anc_right:
+            point.setX(point.x() - self.width())
+
+        # Computing completion widget and its parent bottom positions
+        comp_bottom = point.y() + self.height()
+        ancestor = self.parent()
+        if ancestor is None:
+            anc_bottom = screen_bottom
+        else:
+            anc_bottom = min([ancestor.y() + ancestor.height(), screen_bottom])
+
+        # Moving completion widget above if there is not enough space below
+        x_position = point.x()
+        if comp_bottom > anc_bottom:
+            point = self.textedit.cursorRect().topRight()
+            point = self.textedit.mapToGlobal(point)
+            point.setX(x_position)
+            point.setY(point.y() - self.height())
+
+        if ancestor is not None:
+            # Useful only if we set parent to 'ancestor' in __init__
+            point = ancestor.mapFromGlobal(point)
+        self.move(point)
+
+        if not self.is_internal_console:
+            tooltip_point = point
+            tooltip_point.setX(point.x() + self.width())
+            tooltip_point.setY(point.y() + self.sizeHintForRow(0) + 4)
+            for completion in self.completion_list:
+                completion['point'] = tooltip_point
+
+        # Show hint for first completion element
+        self.setCurrentRow(0)
+        self.row_changed(0)
+
+        # signal used for testing
+        self.sig_show_completions.emit(completion_list)
+
+    def update_list(self, filter_text):
+        """
+        Update the displayed list by filtering self.completion_list.
+
+        If returns False, the autocompletion should stop.
```

Comment:
@dalthviz it is not clear what you mean here. Could you elaborate?

Reply:
@goanpeca I think this was part of PR #9260 but it seems that the docstring is not reflecting the implemented behavior (at least the last part), should I remove it or change it for something like `If no items are left on the list the autocompletion should stop`?

Reply:
Yes :-p